doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,282 @@
+.. _can_interface:
+
+Controller Area Network (CAN)
+#############################
+
+.. contents::
+    :local:
+    :depth: 2
+
+Overview
+********
+
+Controller Area Network is a two-wire serial bus specified by the
+Bosch CAN Specification, Bosch CAN with Flexible Data-Rate specification and the
+ISO 11898-1:2003 standard.
+CAN is mostly known for its application in the automotive domain. However, it
+is also used in home and industrial automation and other products.
+
+A CAN transceiver is an external device that converts the logic level signals
+from the CAN controller to the bus-levels. The bus lines are called
+CAN High (CAN H) and CAN Low (CAN L).
+The transmit wire from the controller to the transceiver is called CAN TX,
+and the receive wire is called CAN RX.
+These wires use the logic levels whereas the bus-level is interpreted
+differentially between CAN H and CAN L.
+The bus can be either in the recessive (logical one) or dominant (logical zero)
+state. The recessive state is when both lines, CAN H and CAN L, at roughly at
+the same voltage level. This state is also the idle state.
+To write a dominant bit to the bus, open-drain transistors tie CAN H to Vdd
+and CAN L to ground.
+The first and last node use a 120-ohm resistor between CAN H and CAN L to
+terminate the bus. The dominant state always overrides the recessive state.
+This structure is called a wired-AND.
+
+.. warning::
+
+   CAN controllers can only initialize when the bus is in the idle (recessive)
+   state for a specific amount of time. Therefore you have to make sure that
+   CAN RX is high, at least for a short time. This is also necessary for
+   loopback mode.
+
+.. image:: can_transceiver.svg
+   :width: 70%
+   :align: center
+   :alt: CAN Transceiver
+
+CAN uses so-called identifiers to identify the frame instead of addresses to
+identify a node.
+This identifier can either have 11-bit width (Standard or Basic Frame) or
+29-bit in case of an Extended Frame. The Zephyr CAN API supports both Standard
+and Extended identifiers concurrently. A CAN frame starts with a dominant
+Start Of Frame bit. After that, the identifiers follow. This phase is called the
+arbitration phase. During the arbitration phase, write collisions are allowed.
+They resolve by the fact that dominant bits override recessive bits.
+Nodes monitor the bus and notice their transmission being overridden
+abort their transmission. It results in the fact that lower number identifiers
```

Comment:
`and abort their transmission.`?

Reply:
Yes. They abort. They may, depending on the config, try again when the bus is in idle state again.